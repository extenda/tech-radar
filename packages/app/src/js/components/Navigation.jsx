import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';

import Icon from './Icon';

const Navigation = (props) => {
  const { quadrant, entry, home, tag } = props;

  if (quadrant || entry || tag) {
    return (
      <nav className="navbar" role="navigation">
        <ul className="breadcrumb">
          <li key="home">
            <Link to="/">Radar</Link>
            <Icon name="angle-right" />
          </li>
          {!entry && !tag && <li key={quadrant.dirname}>{quadrant.name}</li>}
          {entry && (
            <>
              <li key={quadrant.dirname}>
                <Link to={`/${quadrant.dirname}.html`}>{quadrant.name}</Link>
                <Icon name="angle-right" />
              </li>
              <li key={entry.name}>{entry.name}</li>
            </>
          )}
          {tag && <li key={tag}>{tag}</li>}
        </ul>
      </nav>
    );
  }

  return (
    <nav className="navbar" role="navigation">
      <ul className="quadrants">
        <li className="radar-root">
          <Link to="/">Radar</Link>
          <Icon name="rss" />
        </li>
        {home.map((q) => (
          <li key={q.dirname}>
            <Link to={`/${q.dirname}.html`}>{q.name}</Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

const quadrantShape = {
  dirname: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

Navigation.propTypes = {
  quadrant: PropTypes.shape(quadrantShape),
  entry: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }),
  tag: PropTypes.string,
  home: PropTypes.arrayOf(PropTypes.shape(quadrantShape)),
};

Navigation.defaultProps = {
  quadrant: null,
  entry: null,
  tag: null,
  home: null,
};

export default Navigation;
