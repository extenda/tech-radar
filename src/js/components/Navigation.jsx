import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Icon from './Icon';

const Navigation = (props) => {
  const { quadrant, entry, home } = props;

  if (quadrant || entry) {
    return (
      <nav className="navbar" role="navigation">
        <ul className="breadcrumb">
          <li>
            <Link to="/">Tech Radar</Link>
            <Icon name="angle-right" />
          </li>
          {!entry && (
            <li>
              {quadrant.name}
            </li>
          )}
          {entry && (
            <React.Fragment>
              <li>
                <Link to={`/${quadrant.dirname}.html`}>{quadrant.name}</Link>
                <Icon name="angle-right" />
              </li>
              <li>
                {entry.name}
              </li>
            </React.Fragment>
          )}
        </ul>
      </nav>
    );
  }

  return (
    <nav className="navbar" role="navigation">
      <ul className="quadrants">
        <li className="radar-root">
          <Link to="/">Tech Radar</Link>
          <Icon name="rss" />
        </li>
        {home.map(q => (
          <li key={q.dirname}>
            <Link to={`/${q.dirname}.html`}>{q.name}</Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

const quadrantShape = {
  dirname: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

Navigation.propTypes = {
  quadrant: PropTypes.shape(quadrantShape),
  entry: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }),
  home: PropTypes.arrayOf(
    PropTypes.shape(quadrantShape)),
};

Navigation.defaultProps = {
  quadrant: null,
  entry: null,
  home: null,
};

export default Navigation;
