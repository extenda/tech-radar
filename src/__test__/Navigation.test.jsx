import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';
import Navigation from '../js/components/Navigation';

describe('<Navigation />', () => {
  test('It renders home menu', () => {
    const home = [
      { name: 'Test 1', dirname: 'test1' },
      { name: 'Test 2', dirname: 'test2' },
      { name: 'Test 3', dirname: 'test3' },
      { name: 'Test 4', dirname: 'test4' },
    ];
    const component = shallow(<Navigation home={home} />);
    expect(component.find(Link)).toHaveLength(5);
    expect(component).toMatchSnapshot();
  });

  test('It renders quadrant breadcrumbs', () => {
    const component = shallow(<Navigation quadrant={{ name: 'Test1', dirname: 'test1' }} />);
    expect(component.find(Link)).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  test('It renders entry breadcrumbs', () => {
    const component = shallow(
      <Navigation
        quadrant={{ name: 'Test1', dirname: 'test1' }}
        entry={{ name: 'Entry' }}
      />,
    );

    expect(component.find(Link)).toHaveLength(2);
    expect(component).toMatchSnapshot();
  });
});
