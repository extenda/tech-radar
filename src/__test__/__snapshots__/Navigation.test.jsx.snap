// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> It renders entry breadcrumbs 1`] = `
<nav
  className="navbar"
  role="navigation"
>
  <ul
    className="breadcrumb"
  >
    <li>
      <Link
        to="/"
      >
        Tech Radar
      </Link>
      <Icon
        name="angle-right"
        spaceBefore={true}
      />
    </li>
    <li>
      <Link
        to="/test1.html"
      >
        Test1
      </Link>
      <Icon
        name="angle-right"
        spaceBefore={true}
      />
    </li>
    <li>
      Entry
    </li>
  </ul>
</nav>
`;

exports[`<Navigation /> It renders home menu 1`] = `
<nav
  className="navbar"
  role="navigation"
>
  <ul
    className="quadrants"
  >
    <li
      className="radar-root"
    >
      <Link
        to="/"
      >
        Tech Radar
      </Link>
      <Icon
        name="rss"
        spaceBefore={true}
      />
    </li>
    <li
      key="test1"
    >
      <Link
        to="/test1.html"
      >
        Test 1
      </Link>
    </li>
    <li
      key="test2"
    >
      <Link
        to="/test2.html"
      >
        Test 2
      </Link>
    </li>
    <li
      key="test3"
    >
      <Link
        to="/test3.html"
      >
        Test 3
      </Link>
    </li>
    <li
      key="test4"
    >
      <Link
        to="/test4.html"
      >
        Test 4
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Navigation /> It renders quadrant breadcrumbs 1`] = `
<nav
  className="navbar"
  role="navigation"
>
  <ul
    className="breadcrumb"
  >
    <li>
      <Link
        to="/"
      >
        Tech Radar
      </Link>
      <Icon
        name="angle-right"
        spaceBefore={true}
      />
    </li>
    <li>
      Test1
    </li>
  </ul>
</nav>
`;
